components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java

```diff
@@ -0,0 +1,263 @@
+import org.wso2.msf4j.Request;
+import org.wso2.siddhi.query.api.SiddhiApp;
+import org.wso2.siddhi.query.api.definition.AggregationDefinition;
+import org.wso2.siddhi.query.api.definition.TableDefinition;
+import org.wso2.siddhi.query.api.definition.WindowDefinition;
+import org.wso2.siddhi.query.compiler.SiddhiCompiler;
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.siddhi.apps.api.rest.impl.utils.Constants.AGGREGATION;
+import static org.wso2.carbon.siddhi.apps.api.rest.impl.utils.Constants.PROTOCOL;
+import static org.wso2.carbon.siddhi.apps.api.rest.impl.utils.Constants.STORE_ANNOTATION;
+import static org.wso2.carbon.siddhi.apps.api.rest.impl.utils.Constants.TABLE;
+import static org.wso2.carbon.siddhi.apps.api.rest.impl.utils.Constants.URL_HOST_PORT_SEPERATOR;
+import static org.wso2.carbon.siddhi.apps.api.rest.impl.utils.Constants.WINDOW;
+import static org.wso2.carbon.siddhi.apps.api.rest.impl.utils.Constants.WORKER_KEY_GENERATOR;
+
+/**
+ * Implementation of SiddhiApps REST API
+ */
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaMSF4JServerCodegen",
+        date = "2017-11-01T11:26:25.925Z")
+public class SiddhiAppsApiServiceImpl extends SiddhiAppsApiService {
+
+    private static final Logger log = LoggerFactory.getLogger(SiddhiAppsApiServiceImpl.class);
+    private static final String PERMISSION_APP_NAME = "DASH";
+    private static final String VIEW_SIDDHI_APP_PERMISSION_STRING = "DASH.siddhiApp.viewer";
+    private static final Type listType = new TypeToken<List<String>>() { }.getType();
+    private Gson gson = new Gson();
+
+    @Override
+    public Response getSiddhiAppDefinitions(Request request, String id, String appName) {
+        if (getUserName(request) != null && !getPermissionProvider().hasPermission(getUserName(request), new
+                Permission(PERMISSION_APP_NAME, VIEW_SIDDHI_APP_PERMISSION_STRING))) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Insufficient permissions to view Siddhi " +
+                    "Apps").build();
+        }
+        String[] hostPort = id.split(WORKER_KEY_GENERATOR);
+        String workerURI = generateURLHostPort(hostPort[0], hostPort[1]);
+        ConfigReader cf = new ConfigReader();
+        String username = cf.getUserName();
+        String password = cf.getPassword();
+
+        try {
+            feign.Response workerResponse = WorkerServiceFactory.getWorkerHttpsClient(
+                    PROTOCOL + workerURI, username, password).getSiddhiAppContent(appName);
+
+            //Get App Content
+            SiddhiAppContent siddhiAppContent = gson.fromJson(workerResponse.body().toString(), SiddhiAppContent.class);
```

Comment:
What if worker returns response that is not 200/null?

Reply:
Fixed in a8ae221